### PR TITLE
DM-11162: Replace all use of Coord and subclasses with SpherePoint

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: include/lsst/meas/modelfit/UnitSystem.h [hunk 1]
```diff
@@ -48,9 +49,9 @@ struct UnitSystem {
      *  set such that unit flux is the given magnitude.  See @ref modelfitUnits for an explanation
      *  of why we frequently use this system.
      */
-    UnitSystem(afw::coord::IcrsCoord const & position, std::shared_ptr<const lsst::afw::image::Calib> calibIn,
+    UnitSystem(afw::geom::SpherePoint const& position, std::shared_ptr<const lsst::afw::image::Calib> calibIn,
```
Comment (rearmstr) [inline include/lsst/meas/modelfit/UnitSystem.h]: No need to move the ```&```.  Same below.

Comment (r-owen) [reply include/lsst/meas/modelfit/UnitSystem.h]: clang-format; I can make it a separate commit if you want

Comment (rearmstr) [reply include/lsst/meas/modelfit/UnitSystem.h]: My concern was for consistency in the file/package.  Did it flag the other cases in this file?

Comment (r-owen) [reply include/lsst/meas/modelfit/UnitSystem.h]: I can run clang-format on the file if you prefer. Normally I only run it on code I change.

Comment (rearmstr) [reply include/lsst/meas/modelfit/UnitSystem.h]: I prefer to keep the same style within a file, but I'm happy to be overruled if this is not what others want.

Comment (r-owen) [reply include/lsst/meas/modelfit/UnitSystem.h]: Fair enough. I'll run clang-format as a separate commit.

**Pull request merged**